plugins/modules/aci_l4l7_device.py

```diff
@@ -0,0 +1,352 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_l4l7_device
+short_description: Manage L4-L7 Devices (vns:LDevVip, vns:RsALDevToPhysDomP)
+description:
+- Manage L4-L7 Devices.
+options:
+  tenant:
+    description:
+    - Name of an existing tenant.
+    type: str
+    aliases: [ tenant_name ]
+  device:
+    description:
+    - Name of L4-L7 device
+    type: str
+    aliases: [ device_name, logical_device_name ]
+  context_aware:
+    description:
+      - Is device Single or Multi context aware
+    type: str
+    choices: [ multi, single ]
+  dev_type:
+    description:
+    - Device type
+    type: str
+    choices: [ physical, virtual ]
+  func_type:
+    description:
+    - Function type of the device
+    type: str
+    choices: [ None, GoTo, GoThrough, L1, L2 ]
+  is_copy:
+    description:
+    - Is the device a copy device
+    type: bool
+  managed:
+    description:
+    - Is the device a managed device
+    type: bool
+  prom_mode:
+    description:
+    - Enable promiscuous mode
+    type: bool
+  svc_type:
+    description:
+    - Service type running on the device
+    type: str
+    choices: [ adc, fw, others, copy ]
```

Comment:
Do we need copy here as input? Or is this configuration that is deduced from is_copy bool?

Reply:
removed